plenum/server/replica.py

```diff
@@ -901,7 +906,7 @@ def processPrePrepare(self, pre_prepare: PrePrepare, sender: str):
         """
         key = (pre_prepare.viewNo, pre_prepare.ppSeqNo)
         self.logger.debug("{} received PRE-PREPARE{} from {}".format(self, key, sender))
-
+        self._setup_last_ordered_for_non_master(is_catchup=False)
```

Comment:
This call must be moved to the branch handling PP_CHECK_NOT_NEXT error at the point after the call of `enqueue_pre_prepare` due to the following reasons:
1. It must be called only if `ppSeqNo` is not next.
2. It must be called after addition of the incoming message to `prePreparesPendingPrevPP` or `preparesWaitingForPrePrepare`.

In case the prepared certificate is completed and `last_ordered_3pc` is adjusted for it then actual processing of the pending messages (from `prePreparesPendingPrevPP` and `preparesWaitingForPrePrepare`) will be initiated by `dequeue_pre_prepares` call when `Replica.serviceQueues` procedure is executed next time.

Reply:
I think we should not do it because in pre-prepare processing node when it recieves message after catch up (for example), in this time backup replica has not correct last ordered value and we try to reestablish it before processing.